### PR TITLE
Added new parameters ApplicationSecurityGroup, IpConfiguration, CustomNetworkInterfaceName for Get-AzPrivateEndpoint and New-AzPrivateEndpoint

### PR DESCRIPTION
**Pull request by vidai-msft**
## Description

Added properties ApplicationSecurityGroups, IpConfigurations, CustomNetworkInterfaceName for PSPrivateEndpoint object. This new properties will be returned as the output for cmdlet Get-AzPrivateEndpoint as well as an optional input for cmdlet New-AzPrivateEndpoint.

Added new cmdlet New-AzPrivateEndpointIpConfiguration to create IpConfiguration object for building new private endpoint.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/PrivateEndpoint/NewAzurePrivateEndpointCommand.cs [hunk 1]
```diff
@@ -91,6 +92,9 @@ public class NewAzurePrivateEndpoint : PrivateEndpointBaseCmdlet
             HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Application Security Groups")]
+        public PSApplicationSecurityGroup[] ApplicationSecurityGroups { get; set; }
```
Comment (msJinLei) [inline src/Network/Network/PrivateEndpoint/NewAzurePrivateEndpointCommand.cs:96]: Please use a singular name

**Review by msJinLei — CHANGES_REQUESTED** [review]
- Please also help the help doc for Get-AzPrivateEndpoint and New-AzPrivateEndpoint
- Please resolve the conflict

File: src/Network/Network/PrivateEndpoint/NewAzurePrivateEndpointCommand.cs (hunk 1 above)
Comment (msJinLei) [reply src/Network/Network/PrivateEndpoint/NewAzurePrivateEndpointCommand.cs]: @vidai-msft forgot to mention. Please update the help doc for the cmdlet as you add a parameter

**Review by msJinLei — DISMISSED** [review]

**Comment by msJinLei** [comment]
@vidai-msft you can suppress the static analysis error since your new cmdlet doesn't need to support should process.

**Comment by vidai-msft** [comment]
Suppressed ShouldProcess error raised by Static Analysis

**Review by msJinLei — APPROVED** [review]

**Pull request merged**